File: DheerajKumar/problem20.cpp
//20. WAP to perform arithmetic operations as per user's choice.


#include <iostream>
using namespace std;

int arithmeticOperation(int number1, int number2, char operation){
    switch (operation)
    {
    case '+':
        return number1 + number2;
    case '-':
        return max(number1, number2) - min(number1, number2);
    case '*':
        return number1 * number2;
    case '/':
        return max(number1, number2) / min(number1, number2);
    
    default:
        cout<< "Operaton dont exist !";
    }
    return 0;
}

int main(){
    int number1, number2;
    char operation;

    cout<<"Enter number 1 -> ";
    cin>>number1;
    cout<<"Operation -> ";
    cin>>operation;
    cout<<"Enter number 2 -> ";
    cin>>number2;

    cout<< "After operation -> "<<arithmeticOperation(number1, number2, operation);
}

File: LovelyKumari/BasicProgramming/Problem20.java
import java.util.Scanner;

public class Problem20 {
    //WAP to perform arithmetic operations as per user's choice
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.println("Choose an operation: ");
        System.out.println("1. Addition");
        System.out.println("2. Subtraction");
        System.out.println("3. Multiplication");
        System.out.println("4. Division");

        System.out.print("Enter your choice [1/2/3/4]: ");
        int choice=sc.nextInt();
        System.out.print("Enter first number: ");
        int num1=sc.nextInt();
        System.out.print("Enter second number: ");
        int num2=sc.nextInt();

        double ans=0;

        switch (choice) {
            case 1:
                ans=num1+num2;
                System.out.print("Addition result: " + ans);  
                break;
            case 2:
                ans=num1-num2;
                System.out.print("Subtraction reuslt: " + ans );  
                break;
            case 3:
                ans = num1*num2 ;
                System.out.println("Multiplication result: "+ ans);
                break;
            case 4:
                if(num2 != 0){
                    ans=num1/num2 ;
                    System.out.println("Division result: "+ ans);
                } else{
                    System.out.println("Error! Division by zero is not allowed.");
                }
                break ;
            default:
                System.out.println("Invalid choice! Please select a valid operation..");
                break;
        }
    }
    
}
